src/java.base/share/classes/sun/security/util/Debug.java

```diff
@@ -208,16 +269,53 @@ public void println(Object obj, String message)
 
     public void println()
     {
-        System.err.println(prefix + ":");
+        System.err.println(prefix + extraInfo() + ":");
     }
 
     /**
      * print a message to stderr that is prefixed with the prefix.
      */
 
-    public static void println(String prefix, String message)
+    public void println(String prefix, String message)
     {
-        System.err.println(prefix + ": "+message);
+        System.err.println(prefix + extraInfo() + ": " + message);
+    }
+
+    /**
+     * If thread debug option enabled, include information containing
+     * hex value of threadId and the current thread name
+     * If timestamp debug option enabled, include timestamp string
+     * @return extra info if debug option enabled.
+     */
+    private String extraInfo() {
+        String retString = "";
+        if (printThreadDetails) {
+            retString = toHexString(Thread.currentThread().threadId()) + "|" +
+                    Thread.currentThread().getName() + "|" + formatCaller();
```

Comment:
The thread ID is in decimal in the default string representation of thread, also other diagnostic features such as the new thread dump. So I think you want to change this to "#" + Thread.currentThread().threadId().

Reply:
Thanks Alan. Today, the TLS `javax.net.debug` logging prints the thread ID in hex format. I was repeating behaviour seen from that impl. Having `java.security.debug` output in decimal and the other in hex would prove confusing. We could consider flipping both to decimal format in a follow on patch.

Reply:
Change it here, or in a later PR is okay. Main thing is that the logs are confusing then the Thread string is "#<id>" but the TLS debug logs are different.